packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -504,7 +504,7 @@ class _SnackBarState extends State<SnackBar> {
         children: <Widget>[
           Expanded(
             child: Container(
-              padding: const EdgeInsets.symmetric(vertical: _singleLineVerticalPadding),
+              padding: widget.padding == null ? const EdgeInsets.symmetric(vertical: _singleLineVerticalPadding) : null,
```

Comment:
This behavior is not documented anywhere. The documentation of `SnackBar.padding` applies to the SnackBar's contents, why would it affect the outside padding of the SnackBar?

Reply:
> why would it affect the outside padding of the SnackBar?

This is because we removed the `singleLineVerticalPadding` property, https://github.com/flutter/flutter/pull/78133#discussion_r604277207, to not confuse the users on which property can override the default vertical padding.

I forgot to update the documentation of `SnackBar.padding`. If my latest commit is okay, I'll proceed updating the documentation. 🙇

Reply:
> So SnackBar.padding wraps the entire contents of the SnackBar (content & action)

Yes.

> and this would only wrap the SnackBar.content (no action)

Yes. But only a vertical padding, `_singleLineVerticalPadding`.


> So if the user sets SnackBar.padding to go around the content and action, there will be 0 padding in between the content and the action?

The latest commit will only set vertical padding of the content to `null` but there's still a padding in between the content and action (horizontal padding).

Example:
```
SnackBar(
  content: const Text('The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog.'),
  action: SnackBarAction(label: 'Sample Action', onPressed: () {}),
  padding: const EdgeInsets.all(0),
),
```
Will display...
<img width="358" alt="image" src="https://user-images.githubusercontent.com/13028267/118425077-fc8b5100-b6fa-11eb-81df-ee4117f1878c.png">

Reply:
I think that makes sense. Would you like to update the documentation?

Reply:
@Piinks Is this update, 9c4ee1b, okay?